File: Source/iOS/Genealogy/Genealogy/Model/A03/DTO/NoticeDTO.m
//
//  NoticeDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "NoticeDTO.h"

@implementation NoticeDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"noticeID"] isKindOfClass:[NSNull class]]) {
        self.noticeID = jsonData[@"noticeID"];
    }
    if (![jsonData[@"date"] isKindOfClass:[NSNull class]]) {
        self.date = jsonData[@"date"];
    }
    if (![jsonData[@"time"] isKindOfClass:[NSNull class]]) {
        self.time = jsonData[@"time"];
    }
    if (![jsonData[@"distance"] isKindOfClass:[NSNull class]]) {
        self.distance = jsonData[@"distance"];
    }
    if (![jsonData[@"content"] isKindOfClass:[NSNull class]]) {
        self.content = jsonData[@"content"];
    }
    if (![jsonData[@"userFace"] isKindOfClass:[NSNull class]]) {
        self.userFace = jsonData[@"userFace"];
    }
    if (![jsonData[@"userName"] isKindOfClass:[NSNull class]]) {
        self.userName = jsonData[@"userName"];
    }
    if (![jsonData[@"userPhone"] isKindOfClass:[NSNull class]]) {
        self.userPhone = jsonData[@"userPhone"];
    }
    if (![jsonData[@"state"] isKindOfClass:[NSNull class]]) {
        self.state = jsonData[@"state"];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.noticeID) {
        dic[@"noticeID"] = self.noticeID;
    }
    if (self.date) {
        dic[@"date"] = self.date;
    }
    if (self.time) {
        dic[@"time"] = self.time;
    }
    if (self.distance) {
        dic[@"distance"] = self.distance;
    }
    if (self.content) {
        dic[@"content"] = self.content;
    }
    if (self.userFace) {
        dic[@"userFace"] = self.userFace;
    }
    if (self.userName) {
        dic[@"userName"] = self.userName;
    }
    if (self.userPhone) {
        dic[@"userPhone"] = self.userPhone;
    }
    if (self.state) {
        dic[@"state"] = self.state;
    }
    return dic;
}

@end

File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/a03/dto/NoticeDTO.java
﻿/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.a03.dto;

/**
 * NoticeDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/06             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("通知数据")
public class NoticeDTO implements java.io.Serializable, Comparable<NoticeDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private String noticeID = null;

	public String getNoticeID() {
		return this.noticeID;
	}

	public void setNoticeID(
			String noticeID) {
		this.noticeID = noticeID;
	}

	private java.sql.Date date = null;

	public java.sql.Date getDate() {
		return this.date;
	}

	public void setDate(
			java.sql.Date date) {
		this.date = date;
	}

	private java.sql.Time time = null;

	public java.sql.Time getTime() {
		return this.time;
	}

	public void setTime(
			java.sql.Time time) {
		this.time = time;
	}

	private String distance = null;

	public String getDistance() {
		return this.distance;
	}

	public void setDistance(
			String distance) {
		this.distance = distance;
	}

	private String content = null;

	public String getContent() {
		return this.content;
	}

	public void setContent(
			String content) {
		this.content = content;
	}

	private String userFace = null;

	public String getUserFace() {
		return this.userFace;
	}

	public void setUserFace(
			String userFace) {
		this.userFace = userFace;
	}

	private String userName = null;

	public String getUserName() {
		return this.userName;
	}

	public void setUserName(
			String userName) {
		this.userName = userName;
	}

	private String userPhone = null;

	public String getUserPhone() {
		return this.userPhone;
	}

	public void setUserPhone(
			String userPhone) {
		this.userPhone = userPhone;
	}

	private Integer state = null;

	public Integer getState() {
		return this.state;
	}

	public void setState(
			Integer state) {
		this.state = state;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		NoticeDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}
